index.html

```diff
@@ -208,91 +285,136 @@ <h2>
-            already locked (or pending lock) the pointer lock <a>target</a>
-            must be updated to this element and a <a>pointerlockchange</a>
-            event sent.
-          </p>
-          <p>
-            If any element, whose <a>shadow-including root</a> is a different
-            document, is already locked the request must fail and a
-            <a>pointerlockerror</a> event be sent.
-          </p>
-          <p>
-            Once in the locked state the <a>user agent</a> must fire all
-            relevant user generated {{MouseEvent}} events (for example:
-            `mousemove`, `mousedown`, `mouseup`, `click`, and `wheel`)
-            [[ui-events]] to the <a>target</a> of pointer lock, and not fire
-            mouse events to other elements. Events that require the concept of
-            a mouse cursor must not be dispatched (for example: `mouseover`,
-            `mouseout`, `drag`, and `drop`).
-          </p>
-          <p>
-            In the locked state the system mouse cursor must be hidden.
-            Movement and button presses of the mouse must not cause the window
-            to lose focus.
-          </p>
-          <p>
-            Synthetic mouse events created by application script act the same
-            regardless of lock state.
+            RequestPointerLock() will follow these steps
+            <ol>
+              <li>
+                If the the <a>target</a>'s
+                <a>shadow-including root</a> is the <a>active document</a> of a
+                [=Document/browsing context=] which is (or has an <a>ancestor
+                browsing context</a> which is) in focus by a window which is in
+                focus by the operating system's window manager queue <a>pointerlockerror</a> 
+                and <a href="https://heycam.github.io/webidl/#reject">reject</a> the promise with a
+                "{{WrongDocumentError}}" {{DOMException}}.
+              </li>
+              <li>
+                If the pointer is already locked by an element whose <a>shadow-including root</a> is a different
+                document queue <a>pointerlockerror</a> and <a href="https://heycam.github.io/webidl/#reject">reject</a> 
+                the promise with a "{{InUseAttributeError}}" {{DOMException}}.
+              </li>
+              <li>
+                If the pointer is already locked by an element whose <a>shadow-including root</a> is a different
+                document queue <a>pointerlockerror</a> and <a href="https://heycam.github.io/webidl/#reject">reject</a> 
+                the promise with a "{{InUseAttributeError}}" {{DOMException}}.
+              </li>
+              <li>
+                If any element (including this one), whose <a>shadow-including
+                root</a> is same as this element's <a>shadow-including root</a>, is
+                already locked (or pending lock) follow these steps:
```

Comment:
Hmm. This is not quite as explicit as would be ideal. I think the best way to do this would be something like:

- Previously, where you state that the user agent has a (pointer-lock) target, also state that it has a **current pointer-lock options** which is a `PointerLockOptions` dictionary or null.
- Be sure that when you lock/unlock and manage the pointer-lock target, you also set the user agent's pointer-lock options.
- Update the check here to "If the user agent's pointer-lock target's shadow-including root is equal to this's shadow-including root".

... Well, I think there's more cleanup in general, but that's a start.

Reply:
Ok I tried to clean this up a bit.  Let me know what you think.